layer/layer_store.go

```diff
@@ -334,7 +334,10 @@ func (ls *layerStore) get(l ChainID) *roLayer {
 }
 
 func (ls *layerStore) Get(l ChainID) (Layer, error) {
-	layer := ls.get(l)
+	ls.layerL.Lock()
+	defer ls.layerL.Unlock()
```

Comment:
We don't need to use a `defer` here, right? `defers` can slow down small, commonly-used functions by up to 20%.